test/state/mpt.cpp

```diff
@@ -132,12 +127,37 @@ class MPTNode
                                     std::move(br);
     }
 
-    /// Finds the position at witch two paths differ.
-    static size_t mismatch(const Path& p1, const Path& p2) noexcept
+    /// The information how two paths are different.
+    struct MismatchResult
     {
-        assert(p1.length <= p2.length);
-        return static_cast<size_t>(
-            std::mismatch(p1.nibbles, p1.nibbles + p1.length, p2.nibbles).first - p1.nibbles);
+        static constexpr uint8_t invalid = 0xff;
+
+        Path common;                ///< Common prefix.
+        uint8_t nibble1 = invalid;  ///< Mismatched nibble in path 1. Invalid if paths match.
+        uint8_t nibble2 = invalid;  ///< Mismatched nibble in second path.
+        Path tail1;                 ///< Remaining tail after the nibble in path 1.
+        Path tail2;                 ///< Remaining tail after the nibble in path 2.
+    };
+
+    /// Finds the difference between two paths.
+    static MismatchResult mismatch(const Path& p1, const Path& p2) noexcept
+    {
+        assert(p1.length <= p2.length);  // no support for inserting keys of different lengths
```

Comment:
If there's no support for different lengths, why doesn't this assert equality of them?

Reply:
These are not the full paths (the ones being inserted by users) but sub-paths of the nodes. Once a prefix of a path is processes its tails (produced in this function) is used further down the trie.

Reply:
Extended this documentation section as I gathered new information.

Reply:
Still this comment here is confusing. This is not even an inserting function.

Reply:
This assert was too much. It was replaced with a weaker one with extended explanation (hopefully understandable).